crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -115,18 +115,55 @@ type IdCursor = isize;
 /// [`EntityCommands`]: crate::system::EntityCommands
 /// [`Query::get`]: crate::system::Query::get
 /// [`World`]: crate::world::World
-#[derive(Clone, Copy, Eq, Ord, PartialOrd)]
+#[derive(Clone, Copy, Eq)]
+// Alignment repr necessary to allow LLVM to better output
+// optimised codegen for `to_bits`, `PartialEq` and `Ord`.
+// See <https://github.com/rust-lang/rust/issues/106107>
+#[repr(align(8))]
```

Comment:
Adding a field would complicate this PR, as then I'd have to change initialisation usages of `Entity` within the file. It would add either complication, or noise to change things to use `new` instead (which in my view, should be the case with constructors anyway). Also, the only 32-bit platform mainstream enough to worry about is wasm32, which should handle `u64` just fine as it is a native type. All in all, I'd rather not complicate the structure of `Entity` too much. Maybe something for a different PR to explore if there's further potential there (or something to bake into #9797)